handler_test.go

```diff
@@ -0,0 +1,1982 @@
+								{
+									IP:         "1.1.1.1",
+									TTLSeconds: 20,
+								},
+								{
+									IP:         "1.1.1.2",
+									TTLSeconds: 20,
+								},
+							},
+							ResolutionFailures: 0,
+							Conditions: []metav1.Condition{
+								{
+									Type:    ConditionDegraded,
+									Status:  metav1.ConditionFalse,
+									Reason:  dns.RcodeToString[dns.RcodeSuccess],
+									Message: rcodeMessage[dns.RcodeSuccess],
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		dnsQueryTestCases: []test.Case{
+			{
+				Qname: "www.example.com.",
+				Qtype: dns.TypeA,
+				Rcode: dns.RcodeSuccess,
+				Answer: []dns.RR{
+					test.A("www.example.com. 30 IN A 1.1.1.1"),
+					test.A("www.example.com. 30 IN A 1.1.1.2"),
+				},
+			},
+		},
+		expectedStatuses: []ocpnetworkapiv1alpha1.DNSNameResolverStatus{
+			{
+				ResolvedNames: []ocpnetworkapiv1alpha1.DNSNameResolverResolvedName{
+					{
+						DNSName: "www.example.com.",
+						ResolvedAddresses: []ocpnetworkapiv1alpha1.DNSNameResolverResolvedAddress{
+							{
+								IP:         "1.1.1.1",
+								TTLSeconds: 30,
+							},
+							{
+								IP:         "1.1.1.2",
+								TTLSeconds: 30,
+							},
+						},
+						ResolutionFailures: 0,
+						Conditions: []metav1.Condition{
+							{
+								Type:    ConditionDegraded,
+								Status:  metav1.ConditionFalse,
+								Reason:  dns.RcodeToString[dns.RcodeSuccess],
+								Message: rcodeMessage[dns.RcodeSuccess],
+							},
+						},
+					},
+				},
+			},
+		},
+	},
+	{
+		name: "Don't update the ttl of an IP in the regular dns name resolver object's status if next lookup time hasn't changed",
+		dnsNameResolvers: []ocpnetworkapiv1alpha1.DNSNameResolver{
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "regular",
+					Namespace: "dns",
+				},
+				Spec: ocpnetworkapiv1alpha1.DNSNameResolverSpec{
+					Name: "www.example.com.",
+				},
+				Status: ocpnetworkapiv1alpha1.DNSNameResolverStatus{
+					ResolvedNames: []ocpnetworkapiv1alpha1.DNSNameResolverResolvedName{
+						{
+							DNSName: "www.example.com.",
+							ResolvedAddresses: []ocpnetworkapiv1alpha1.DNSNameResolverResolvedAddress{
+								{
+									IP:         "1.1.1.1",
+									TTLSeconds: 30,
+								},
+								{
+									IP:         "1.1.1.2",
+									TTLSeconds: 30,
+								},
+							},
+							ResolutionFailures: 0,
+							Conditions: []metav1.Condition{
+								{
+									Type:    ConditionDegraded,
+									Status:  metav1.ConditionFalse,
+									Reason:  dns.RcodeToString[dns.RcodeSuccess],
+									Message: rcodeMessage[dns.RcodeSuccess],
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		dnsQueryTestCases: []test.Case{
+			{
+				Qname: "www.example.com.",
+				Qtype: dns.TypeA,
+				Rcode: dns.RcodeSuccess,
+				Answer: []dns.RR{
+					test.A("www.example.com. 30 IN A 1.1.1.1"),
+					test.A("www.example.com. 30 IN A 1.1.1.2"),
```

Comment:
The solution proposed in https://github.com/openshift/coredns-ocp-dnsnameresolver/pull/1#discussion_r1526163769 can be used so that we can have a different TTL here in the query.